python/src/aac/lang/language_context.py

```diff
@@ -219,12 +222,11 @@ def update_definition_in_context(self, definition: Definition):
             definition (Definition): The Definition to update in the context.
         """
 
-        if definition.uid in self.definitions_dictionary:
-            old_definition = self.definitions_dictionary.get(definition.uid)
-
+        old_definition = self.definitions_dictionary.get(definition.uid)
+        if old_definition and self._is_definition_editable(definition):
```

Comment:
Similar to my comment above, we should also give a specific error message here for the case when someone tries to update a non-editable definition (i.e. `Primitives`, etc).